pkg/operator/api/api.go

```diff
@@ -63,72 +62,81 @@ type apiServer struct {
 	// notify all dapr runtime
 	connLock          sync.Mutex
 	allConnUpdateChan map[string]chan *componentsapi.Component
+	ready             chan struct{}
 }
 
 // NewAPIServer returns a new API server.
 func NewAPIServer(client client.Client) Server {
 	return &apiServer{
 		Client:            client,
 		allConnUpdateChan: make(map[string]chan *componentsapi.Component),
+		ready:             make(chan struct{}),
```

Comment:
Here too IMHO a sync.Atomic.Bool and then using CompareAndSwap could make a better semaphore

Reply:
If we change ready to an atomic bool, there is no way for `Ready` to block until the api is ready or the context is cancelled.